components/Tags.tsx

```diff
@@ -1,53 +1,98 @@
+
+export interface TagData {
+  readonly tagName: string;
+  readonly postsCount: number;
+}
 
 interface Props {
-  uniqueTags: string[];
+  readonly sortedTags: readonly TagData[];
 }
+interface TagItemProps {
+  readonly isActive: boolean;
+  readonly onClick: () => void;
+  readonly tagName: string;
+  readonly tagCount?: number;
+}
+
+interface TagsListProps {
+  readonly tagsData: readonly TagItemProps[];
+}
+
+const TagItem = ({
+  isActive,
+  onClick,
+  tagName,
+  tagCount,
+}: TagItemProps): ReactElement => {
+  return (
+    <li className={isActive ? "active" : ""} onClick={onClick}>
+      {tagCount ? `${tagName} [${tagCount}]` : tagName}
+    </li>
+  );
+};
+
+const TagsList = ({ tagsData }: TagsListProps): ReactElement => {
+  return (
+    <ul className="filter-tags-list">
+      {tagsData.map((tagData) => (
+        <TagItem key={tagData.tagName} {...tagData} />
+      ))}
+    </ul>
+  );
+};
 
-const Tags = ({ uniqueTags }: Props) => {
+const Tags = ({ sortedTags }: Props) => {
   const selectedTags = useAppSelector((state) => state.selectedTags);
 
-  const dispatch = useAppDispatch();
+  const { edit } = useEditState();
 
-  const noneTagSelected = selectedTags.length === 0;
-  const allTagsSelected = uniqueTags.length === selectedTags.length;
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
     dispatch(resetPaginationPage());
   }, [dispatch, selectedTags]);
 
-  return (
-    <ul className="filter-tags-list">
-      <li
-        className={noneTagSelected || allTagsSelected ? "active" : ""}
-        onClick={() => dispatch(resetTags())}
-      >
-        ALL
-      </li>
-      {uniqueTags.map((uniqueTag) => (
-        <li
-          key={uniqueTag}
-          className={selectedTags.includes(uniqueTag) ? "active" : ""}
-          onClick={() => {
-            if (selectedTags.includes(uniqueTag)) {
-              const updatedTags = selectedTags.filter(
-                (tag) => tag !== uniqueTag
-              );
-
-              dispatch(setTags(updatedTags));
-            } else {
-              dispatch(setTags([...selectedTags, uniqueTag]));
-            }
-          }}
-        >
-          {uniqueTag}
-        </li>
-      ))}
-    </ul>
-  );
+  const tagSelect = (uniqueTag: string): void => {
+    if (selectedTags.includes(uniqueTag)) {
+      const updatedTags = selectedTags.filter((tag) => tag !== uniqueTag);
+
+      dispatch(setTags(updatedTags));
+    } else {
+      dispatch(setTags([...selectedTags, uniqueTag]));
+    }
+  };
+
+  const tagsData: readonly TagItemProps[] = sortedTags
+    .map(({ tagName, postsCount }) => {
+      const isBasicTag = postsCount >= 2;
+      const isAdditionalTag =
+        postsCount === 1 && selectedTags.includes(tagName);
+
+      return edit || isAdditionalTag || isBasicTag
+        ? {
+            isActive: selectedTags.includes(tagName) || isAdditionalTag,
+            tagName,
+            tagCount: edit ? postsCount : undefined,
```

Comment:
I think we have a rule against a nested ternary

Does this code work?

Reply:
Yes, because there is no nested ternary